br/pkg/lightning/backend/local/local.go

```diff
@@ -1681,6 +1682,10 @@ func (local *local) ResolveDuplicateRows(ctx context.Context, tbl table.Table, t
 				if err == nil {
 					return nil
 				}
+				if types.ErrBadNumber.Equal(err) {
+					logger.Warn("delete duplicate rows encounter error", log.ShortError(err))
+					return common.ErrInconsistentSchema.GenWithStackByArgs(tableName)
```

Comment:
from `ErrBadNumber` we cannot conclude there're inconsistent schema, although it's the case we met.

prefer we  name it `ErrResolveDuplicateRows` and add the guess in extra err message

Reply:
I check the codes where return ``ErrBadNumber`` error. This comes from when we assume the data type should be ``Decimal`` but it is not. So I think type inconsistent or table schema inconsistent may be the raw cause. But for sure, I think your suggestion is legitimate.

Besides, there are still some errors we did not handle in this infinite loop. For example: https://github.com/pingcap/tidb/blob/1436ab9e7b4dc9fef3ab4ac359a6c0ae3452c453/util/codec/decimal.go#L53 I suggest modifying it to ``ErrBadNumber`` as well.

Reply:
Although not related to our issue, I worried that some potential problems may occur. Therefore, I think extra reviews are appreciated.

Reply:
> Besides, there are still some errors we did not handle in this infinite loop

we can do it in later pr